src/box/lua/console.lua

```diff
@@ -401,7 +401,14 @@ local function local_eval(storage, line)
     if not fun then
         return format(false, errmsg)
     end
-    return format(pcall(fun))
+    local in_txn_before = ffi.C.box_txn()
+    -- with pack/unpack there is issue with trailing nils in multiret
+    return (function(success, ...)
+        if not success and not in_txn_before and ffi.C.box_txn() then
+            box.rollback()
+        end
+        return format(success, ...)
+    end)(pcall(fun))
```

Comment:
Nit: We generally avoid creating closures on demand due to extra Lua GC pressure, when it is possible to define a function and call it many times.

Reply:
Ok. Closure unnecessary here in the first place. I failed at first to pass multiret with trailing nils with `unpack({pcall()})`. For example:

```
 tarantool> (function(...) print(unpack{...}) end)(1, nil)
1
---
...
```

They say `table.pack` available in Lua 5.2 allows to handle this situation but we don't have it. Fortunately one can write table.pack in Lua! :)

Reply:
I see, you suggested introducing a function and I considered as alternative to pack/unpack as in latest version.